### DISCUSSION TITLE
Should kvantum-dark be mentioned?

### DISCUSSION THREAD

**Discussion by francoism90** [tsujan/Kvantum#728] [Q&A] [answered]
I'm using Kvantum on GS to style Qt apps, which works really great. :)

However, when using `QT_STYLE_OVERRIDE=kvantum` on dark themes, it breaks text colors/make them unreadable.
It does work fine when using `QT_STYLE_OVERRIDE=kvantum-dark` instead.

Is this correct behavior? Should it be added to INSTALL.md?

Thanks!

**Comment by tsujan** [comment, marked as answer]
`kvantum-dark` was made  for Gnome; otherwise, it's redundant and you could ignore it. It just switches to the theme "XDark" if the theme "X" has been chosen and "XDark" exists.

For example, if the active theme is KvGnome, `kvantum-dark` switches to KvGnomeDark.

Kvantum doesn't break text colors; bad hard-coded styles in apps may do it. Also, a theme that's made for Kvantum may have issues, but no theme that comes with Kvantum has such an issue.

**Reply by tsujan** [reply]
BTW, `kvantum-dark` is already explained in [Theme-Making.pdf](https://github.com/tsujan/Kvantum/blob/master/Kvantum/doc/Theme-Making.pdf) → Dark variants of themes

**Reply by francoism90** [reply]
@tsujan Thanks for your reply and the information how the dark themes work.

I'll stick with `kvantum-dark` for now, as this work with Qogir-dark (non provided theme).

**Reply by tsujan** [reply]
You're welcome.

> I'll stick with kvantum-dark for now

If your active theme is called "Qogir-dark" and there's no installed theme with the name "Qogir-darkDark", `kvantum` and `kvantum-dark` will be exactly the same.

**Reply by francoism90** [reply]
When using `QT_STYLE_OVERRIDE=kvantum-dark`:

![image](https://user-images.githubusercontent.com/5028905/143441382-15e954dc-3c66-4d4a-9cdb-9cd6094c5478.png)

When using `QT_STYLE_OVERRIDE=kvantum`:

![image](https://user-images.githubusercontent.com/5028905/143441493-c959162c-ef43-4b73-98a2-0499a37c70a7.png)

This also happens with inbuilt themes and `QT_STYLE_OVERRIDE=kvantum`:

![image](https://user-images.githubusercontent.com/5028905/143441740-d8ce4e0b-c277-4291-ae59-0004820aae24.png)

It also doesn't work when using the default GTK-theme Adwaita. But everything does work fine when using `QT_STYLE_OVERRIDE=kvantum-dark`.

Is it something GTK/GS related?

**Reply by tsujan** [reply]
Do you have `qt5ct` installed and set?

I guess you work under a GTK DE like Gnome. GTK DEs usually don't have a Qt plugin (= "Qt platform integration program", as it's called in Kvantum Manager); so, you need to install one. `qt5ct` should work everywhere if it's set up correctly.

Anyway, in this special case, `kvantum-dark` tells Gnome that this is a dark theme but it won't be needed if there's a Qt plugin.

**Reply by francoism90** [reply]
Correct, I'm running Gnome(-shell) 41.

I'm using this one: https://wiki.archlinux.org/title/Uniform_look_for_Qt_and_GTK_applications#QGnomePlatform

I've tried using `qt5ct`, but this doesn't seem to apply any theme settings like colors correctly. That's why I'm using Kvantum instead.

**Reply by tsujan** [reply]
> I've tried using qt5ct, but this doesn't seem to apply any theme settings like colors correctly

I'm afraid `qt5ct` is a platform plugin, not a style. It enables you to choose Kvantum or any other style (Kvantum has its own internal colors).

I think `QGnomePlatform` may have a problem or be too elementary or too limited. I remember that a Fedora user reported a similar issue but his problem was solved after installing and setting up `qt5ct` and also choosing Kvantum in it.

Unfortunately, Gnome is unfriendly toward Qt. In Qt DEs, like KDE or LXQt, the user doesn't need to know about "platform plugin", "style plugin" and their difference because everything needed is already installed.

I explain them briefly:

Since Qt5, *a platform plugin* is needed for choosing the Qt style and also the icon theme that's used by Qt apps. A platform plugin may also be called "Qt configuration plugin". Some platform plugins, like `qt5ct`, have a GUI that enables the user to select the icon theme and Qt style.

*Qt style* is sometimes called "Qt style plugin" or "Qt widget style". It's responsible for styling Qt apps. The default style of Qt is called Fusion but there are other styles like Kvantum, QtCurve, Breeze, etc. For changing the style, a platform plugin is needed, preferably with a nice GUI (like that of `qt5ct`).
